app/src/main/java/org/mozilla/tv/firefox/utils/PeriodicRequester.kt

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.utils
+
+import android.support.annotation.VisibleForTesting
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import java.util.concurrent.TimeUnit
+import kotlin.math.pow
+
+private const val CACHE_UPDATE_FREQUENCY_MINUTES = 45L
+private val CACHE_UPDATE_FREQUENCY_SECONDS = TimeUnit.MINUTES.toSeconds(CACHE_UPDATE_FREQUENCY_MINUTES)
+
+/**
+ * Simple interface for making a network request.
+ *
+ * This may operate on any thread, and consumers should take care to schedule
```

Comment:
This is a bold claim, can we use annotations?

Reply:
Sounds like a good idea.  What would be the correct annotation for that?  I'm not sure.